src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,82 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    internal class HealthHeartbeatDefaultPayload : IHealthHeartbeatPayloadExtension
+    {
+        public const string FieldRuntimeFrameworkVer = "runtimeFramework";
+        public const string FieldTargetFramework = "targetFramework";
+        public const string FieldAppInsightsSdkVer = "appinsightsSdkVer";
+
+        public string EnabledProperties;
+
+        public HealthHeartbeatDefaultPayload(string enabledProperties)
+        {
+            this.EnabledProperties = enabledProperties;
+        }
+
+        public string Name => "SDKHealthHeartbeat";
+
+        public int CurrentUnhealthyCount
+        {
+            get { return 0; }
+        }
+
+        public IEnumerable<KeyValuePair<string, object>> GetPayloadProperties()
+        {
+            var payload = new Dictionary<string, object>();
+            if (this.IsFieldEnabled(FieldTargetFramework))
+            {
+                payload.Add(FieldTargetFramework, this.GetTargetFrameworkVer());
+            }
+
+            if (this.IsFieldEnabled(FieldAppInsightsSdkVer))
+            {
+                payload.Add(FieldAppInsightsSdkVer, this.GetAppInsightsSdkVer());
+            }
+
+            if (this.IsFieldEnabled(FieldRuntimeFrameworkVer))
+            {
+                payload.Add(FieldRuntimeFrameworkVer, this.GetRuntimeFrameworkVer());
+            }
+
+            return payload;
+        }
+
+        private bool IsFieldEnabled(string fieldName)
+        {
+            if (!this.EnabledProperties.Equals("*", StringComparison.OrdinalIgnoreCase))
```

Comment:
if you call it on every report of telemetry - I'd suggest to optimize this logic. As mentioned above use any sort of list instead of string

Reply:
Removed this logic. Will revisit the polling design I've created and update it to more of an updatable state machine idea (consumers of the SDK 'push' new state information as it changes rather than us polling for it every interval).